pkg/vm/memory/memory.go

```diff
@@ -41,17 +58,30 @@ func (segment *Segment) Read(index f.Element) *MemoryValue {
 // todo(rodro): Check out temprary segments
 // Represents the whole VM memory divided into segments
 type Memory struct {
-	Segments []Segment
+	Segments []*Segment
 }
 
 // todo(rodro): can the amount of segments be known before hand?
-func InitializeEmptyMemory() Memory {
-	return Memory{
+func InitializeEmptyMemory() *Memory {
+	return &Memory{
 		// size 4 should be enough for the minimum amount of segments
-		Segments: make([]Segment, 4),
+		Segments: make([]*Segment, 4),
 	}
 }
 
+func (memory *Memory) LoadBytecode(bytecode *[]f.Element) error {
```

Comment:
Maybe create a general purpose function that fills a segment with an array of felts?